server/src/main/java/io/littlehorse/server/streams/topology/core/processors/CommandProcessor.java

```diff
@@ -93,19 +96,17 @@ private void processHelper(final Record<String, Command> commandRecord) {
                 "{} Processing command of type {} with commandId {} with partition key {}",
                 config.getLHInstanceName(),
                 command.type,
-                command.commandId,
+                command.getCommandId(),
                 command.getPartitionKey());
         try {
             Message response = command.process(executionContext, config);
             executionContext.endExecution();
+            CompletableFuture<Message> completable =
```

Comment:
Please guard this inside the `if ()`. We shouldn't be creating `CompletableFuture`s if there is nothing waiting for a response